cluster/examples/kubernetes/cassandra/operator.yaml

```diff
@@ -0,0 +1,227 @@
+spec:
+  group: cassandra.rook.io
+  names:
+    kind: Cluster
+    listKind: ClusterList
+    plural: clusters
+    singular: cluster
+  scope: Namespaced
+  version: v1alpha1
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          type: object
+          properties:
+            version:
+              type: string
+              # Only allow 3.x.x versions
+              pattern: ^3(\.[0-9]{1,2}){0,2}$
+              description: "Version of Cassandra"
+            datacenter:
+              type: object
+              properties:
+                name:
+                  type: string
+                  description: "Datacenter Name"
+                racks:
+                  type: array
+                  properties:
+                    name:
+                      type: string
+                    members:
+                      type: integer
+                    configMapName:
+                      type: string
+                    storage:
+                      type: object
+                      properties:
+                        volumeClaimTemplates:
+                          type: object
+                          # TODO: Check if we can ref the already existing schema
+                      required:
+                       - "volumeClaimTemplates"
+                    placement:
+                      type: object
+                    resources:
+                      type: object
+                      properties:
+                        # TODO: Check if we can ref the already existing schema
+                        cassandra:
+                          type: object
+                        sidecar:
+                          type: object
+                      required:
+                        - "cassandra"
+                        - "sidecar"
+                    sidecarImage:
+                      type: object
+                  required:
+                    - "name"
+                    - "members"
+                    - "storage"
+                    - "resources"
+              required:
+                - "name"
+          required:
+            - "version"
+            - "datacenter"
+
+---
+
+## Cassandra Member CRD
+#apiVersion: apiextensions.k8s.io/v1beta1
+#kind: CustomResourceDefinition
+#metadata:
+#  name: members.cassandra.rook.io
+#spec:
+#  group: cassandra.rook.io
+#  names:
+#    kind: Member
+#    listKind: MemberList
+#    plural: members
+#    singular: member
+#  scope: Namespaced
+#  version: v1alpha1
+
+---
+
+# ClusterRole for cassandra-operator.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: rook-cassandra-operator
+rules:
+  - apiGroups:
+    - ""
+    resources:
+      - pods
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - services
+      - serviceaccounts
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+  - apiGroups:
+      - rbac.authorization.k8s.io
```

Comment:
i may see it later on in this review pass, but why does the cassandra operator need permissions to create new RBAC entries?  the ceph operator started out that way too but moved to a model where the admin can up front define all the RBAC roles that will be needed, removing higher security permissions from the operator and putting it in the hands of the cluster admin.  Did you consider that approach that Ceph uses as well?

Reply:
@jbw976 That's an interesting proposal. 
On the one hand, the operator already has permissions to do a lot of stuff and can probably wreck havoc if it wants to.
On the other hand, by removing the permission to create RBAC entries, we limit its access to the cluster's Secrets. 

Not too sure what the best approach is here. 
Have you had any complaints/concerns from people using rook?
Maybe you can also point me to the discussion from when rook decided to move those permissions out of the operator.

Reply:
The operator shouldn't need access to the rbac API group to get secrets, can you help me understand that better? Is the operator actively creating new service acocunts, roles, or role bindings? can you point me to where it is doing that and for what reason?